### PR TITLE
Make docker image clear optional

### PR DESCRIPTION
**Pull request by sqrtofsaturn**
We made the docker image clear optional, since we're using docker through the sock and would like to maintain the cache for future builds. We also added an exit 0 after the docker rmi statement at the end because its exit code was being used as the exit code of the entire script, which was affecting one of our builds.

### CODE REVIEW

**Comment by fermayo** [comment]
Actually, that cleanup code was done on one of the first versions of the image, and it was to reduce the amount of trash it left behind in `/var/lib/docker` (which is a volume and never got deleted).
Now, because it can also reuse a docker socket from outside, it doesn't make sense to remove all images like that (could potentially remove what it's not supposed to remove).

I would change the code so it only removes the image that just got built (so we don't need the extra flag) and shows a warning if it fails:

```
docker rmi $IMAGE_NAME > /dev/null 2>&1 || (echo "   WARNING: Failed to delete image")
```

And what I would include is an optional cleanup after the tests to remove the images used in the tests:

```
cat ./${TEST_FILENAME} | grep "image:" | awk '{print $2}' | xargs -n1 docker rmi
```

What do you think?

**Comment by royvandewater** [comment]
I added the last image bit you mentioned:

```
docker rmi $IMAGE_NAME > /dev/null 2>&1 || (echo "   WARNING: Failed to delete image")
```

I left the SKIP_CLEANUP stuff in there, since in our case, the docker thats building the image is also the one thats about to pull it down. 

I'm not sure about the test stuff you mentioned, haven't had much chance to play with test images yet :-)

File: build.sh [hunk 1]
```diff
@@ -84,7 +84,10 @@ if [ ! -z "$IMAGE_NAME" ]; then
 	if [ ! -z "$USERNAME" ] || [ ! -z "$DOCKERCFG" ] || [ -f /.dockercfg ]; then
 		echo "=>  Pushing image $IMAGE_NAME"
 		docker push $IMAGE_NAME
-		docker rmi -f $(docker images -q --no-trunc -a) > /dev/null 2>&1
+		if [ "$SKIP_CLEAN" != "true" ]; then
+			echo "=> Clearing images"
+		  docker rmi $IMAGE_NAME > /dev/null 2>&1 || (echo "   WARNING: Failed to delete image")
```
Comment (fermayo) [inline build.sh:89]: shouldn't this be `docker rmi -f $(docker images -q --no-trunc -a)` as before?

**Comment by fermayo** [comment]
I have added this in https://github.com/tutumcloud/docker-builder/commit/07dea72ec9a5a7203bc26ddfa198ff71b03b1a4f

**Pull request closed without merging**